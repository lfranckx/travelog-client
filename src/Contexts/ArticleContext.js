/*eslint semi: ["error", "always"]*/
import React, { Component } from 'react';

export const nullArticle = {
    author: {},
    tags: [],
};
export const nullAuthor = {};
export const nullUser = {};
export const nullUsersArticles = [];

const ArticleContext = React.createContext({
    error: null,
    article: nullArticle,
    author: nullAuthor,
    user: nullUser,
    articlesList: [],
    authorsList: [],
    usersArticles: nullUsersArticles,
    setError: () => {},
    clearError: () => {},
    setArticle: () => {},
    clearArticle: () => {},
    setAuthor: () => {},
    clearAuthor: () => {},
    setUser: () => {},
    clearUser: () => {},
    setArticlesList: () => {},
    setAuthorsList: () => {},
    setUsersArticles: () => {},
    clearUsersArticles: () => {}
});
export default ArticleContext;

export class ArticleProvider extends Component {
    state = {
        error: null,
        article: nullArticle,
        author: nullAuthor,
        user: nullUser,
        articlesList: [],
        authorsList: [],
        usersArticles: []
    };

    setError = error => {
        console.error(error);
        this.setState({ error });
    };

    clearError = () => {
        this.setState({ error: null });
    };

    setArticle = article => {
        this.setState({ article });
    };

    clearArticle = () => {
        this.setArticle(nullArticle);
    };

    setAuthor = author => {
        this.setState({ author });
    };

    clearAuthor = () => {
        this.setAuthor(nullAuthor);
    };

    setUser = user => {
        console.log('setting user:', user);
        this.setState({ user });
    };

    clearUser = () => {
        this.setState(nullUser);
    };

    setArticlesList = articlesList => {
        this.setState({ articlesList });
    };

    setAuthorsList = authorsList => {
        this.setState({ authorsList });
    };

    setUsersArticles = usersArticles => {
        this.setState({ usersArticles });
    };

    clearUsersArticles = () => {
        this.setUsersArticles(nullUsersArticles);
    };

    render() {
        const value = {
            error: this.state.error,
            article: this.state.article,
            author: this.state.author,
            user: this.state.user,
            articlesList: this.state.articlesList,
            authorsList: this.state.authorsList,
            usersArticles: this.state.usersArticles,
            setError: this.setError,
            clearError: this.clearError,
            setArticle: this.setArticle,
            clearArticle: this.clearArticle,
            setAuthor: this.setAuthor,
            clearAuthor: this.clearAuthor,
            setUser: this.setUser,
            clearUser: this.clearUser,
            setArticlesList: this.setArticlesList,
            setAuthorsList: this.setAuthorsList,
            setUsersArticles: this.setUsersArticles,
            clearUsersArticles: this.clearUsersArticles
        };
        
        return (
            <ArticleContext.Provider value={value}>
                {this.props.children}
            </ArticleContext.Provider>
        );
    }   
}