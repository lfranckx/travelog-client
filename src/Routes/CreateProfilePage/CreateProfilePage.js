import React, { Component } from 'react';
import '../EditProfilePage/EditProfilePage.css';
import CreateProfileForm from '../../Components/CreateProfileForm/CreateProfileForm.js';
import ArticleContext from '../../Contexts/ArticleContext';

export default class CreateProfilePage extends Component {  
    static defaultProps = {
        location: {},
        history: {
            push: () => {}
        },
        match: { params: {} }
    }

    static contextType = ArticleContext;

    handleSubmitForm = () => {
        const { location, history } = this.props;
        const destination = (location.state || {}).from || `/`;
        history.push(destination);
    }
    
    render() {       
        return (
            <section className="edit-profile-section">
                <h2>Create a profile</h2>
                <CreateProfileForm 
                    onSubmitForm={this.handleSubmitForm}
                />
            </section>
        )
    }
}