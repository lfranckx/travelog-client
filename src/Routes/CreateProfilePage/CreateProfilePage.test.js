import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CreateProfilePage from './CreateProfilePage';
import { ArticleProvider } from '../../Contexts/ArticleContext';


describe(`CreateProfilePage component`, () => {
    it(`renders page without crashing`, () => {
        const wrapper = shallow(
            <ArticleProvider>
                <CreateProfilePage />
            </ArticleProvider>
        )
        expect(toJson(wrapper)).toMatchSnapshot()
    });    
})