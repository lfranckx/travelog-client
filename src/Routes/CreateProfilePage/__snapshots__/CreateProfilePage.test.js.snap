// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateProfilePage component renders page without crashing 1`] = `
<ContextProvider
  value={
    Object {
      "active": false,
      "addComment": [Function],
      "article": Object {},
      "articlesList": Array [],
      "author": Object {},
      "authorsList": Array [],
      "clearArticle": [Function],
      "clearAuthor": [Function],
      "clearError": [Function],
      "clearUser": [Function],
      "clearUsersArticles": [Function],
      "comments": Array [],
      "error": null,
      "search": "",
      "setArticle": [Function],
      "setArticlesList": [Function],
      "setAuthor": [Function],
      "setAuthorsList": [Function],
      "setComments": [Function],
      "setError": [Function],
      "setUser": [Function],
      "setUsersArticles": [Function],
      "toggleActive": [Function],
      "updateSearch": [Function],
      "user": Object {},
      "usersArticles": Array [],
    }
  }
>
  <CreateProfilePage
    history={
      Object {
        "push": [Function],
      }
    }
    location={Object {}}
    match={
      Object {
        "params": Object {},
      }
    }
  />
</ContextProvider>
`;
